analyze/src/semantic/expression.rs

```diff
@@ -99,8 +99,7 @@ pub fn analyze(
                             errors.push(Error::MissingArgument(*parameter.id()));
                         }
 
-                        // this case is not possible
-                        (None, None) => (),
+                        (None, None) => break,
```

Comment:
it is possible, I think I let chatGPT convince me into believing zipping two iterators chained with `repeat` would result in a finite result (it did not)

now breaking if both inputs are `None` (indicating both the original lists have been depleted)